dlt/destinations/dataset/utils.py

```diff
@@ -144,6 +145,8 @@ def normalize_query(
                 if len(expanded_path) == 3:
                     if node.db != expanded_path[0]:
                         node.set("catalog", sqlglot.to_identifier(expanded_path[0], quoted=False))
+        if isinstance(node, sge.Alias):
+            node.set("alias", naming.normalize_identifier(node.alias))
```

Comment:
Here I make sure aliases that are created by the user or automatically by ibis in the compile step above adhere to the naming convention of the current destination. The alternative would be to use a different compiler for every destination. Normalizing the alias in the ModelNormalizer step is too late, since this code is also used for just accessing the data. Without this change, `def test_row_counts` will fail for bigquery because an alias with invalid symbols remains in the query. All other destinations seem to accept this.

Reply:
I have reverted this change and am now using the bigquery compiler only for bigquery destinations. Maybe we should be using the correct compiler for each destination type, but it seems duckdb works for all. I am not sure about this one.